source/NVDAObjects/__init__.py

```diff
@@ -1267,3 +1268,35 @@ def getSelectedItemsCount(self,maxCount=2):
 		For performance, this method will only count up to the given maxCount number, and if there is one more above that, then sys.maxint is returned stating that many items are selected.
 		"""
 		return 0
+
+	reviewBounded = False
+	
+	@script(gesture="kb:NVDA+o",
+		# Hardcoded to avoid circular import
+		category=_(u"Text review"),
+		# Translators: A gesture description.
+		description=_(u"Toggles whether review is constrained to the currently visible text"))
+	def script_toggleReviewBounds(self, gesture):
+		try:

```

Comment:
Make sure to start the try block only when you are sure that you're catching the right NotImplementedError, and directly catch it afterwards. I think that would look like:

```
		try:
			outOfBounds = rp.isOutOfBounds()
		except NotImplementedError:
			ui.message(
				# Translators: Reported when review bound configuration isn't supported for this object.
				_(u"Not supported here"))
			return
```